server/paymentProviders/stripe/webhook.ts

```diff
@@ -564,78 +563,92 @@ async function paymentMethodAttached(event: Stripe.Response<Stripe.Event>) {
   });
 }
 
-export const webhook = async (request: Request<unknown, Stripe.Event>) => {
-  const requestBody = request.body;
+async function validateWebhook(request: Request<unknown, Stripe.Event>): Promise<Stripe.Event> {
+  const event = <Stripe.Event>request.body;
+  let stripeAccount = event.account;
+  if (!stripeAccount) {
+    stripeAccount = config.stripe.accountId;
+  }
 
-  debug(`Stripe webhook event received : ${request.rawBody}`);
+  const stripeConnectedAccount = await models.ConnectedAccount.findOne({
+    where: {
+      username: stripeAccount,
+      service: Service.STRIPE,
+    },
+  });
 
-  // Stripe sends test events to production as well
-  // don't do anything if the event is not livemode
-  // NOTE: not using config.env because of ugly tests
-  if (process.env.OC_ENV === 'production' && !requestBody.livemode) {
-    return Promise.resolve();
+  if (!stripeConnectedAccount) {
+    throw new Error(`Stripe connected account not found: ${stripeAccount}`);
   }
 
-  const stripeEvent = {
-    signature: request.headers['stripe-signature'],
-    rawBody: request.rawBody,
-  };
+  let webhookSigningSecret =
+    stripeConnectedAccount.data?.webhookSigningSecret || stripeConnectedAccount.data.stripeEndpointSecret;
 
-  if (requestBody.type === 'issuing_authorization.request') {
-    return virtualcard.processAuthorization(requestBody.data.object, stripeEvent);
+  if (process.env.NODE_ENV === 'development' && !webhookSigningSecret && process.env.STRIPE_WEBHOOK_SIGNING_SECRET) {
+    webhookSigningSecret = process.env.STRIPE_WEBHOOK_SIGNING_SECRET;
```

Comment:
Ideally, we would move all events to the platform account and use only the signing secret in our env.
We can merge this and work on https://github.com/opencollective/opencollective/issues/6203 later.